### PR TITLE
chore: finish wiring up unwrapped keys

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

fixes: https://github.com/confluentinc/ksql/issues/6209

This finishes the wiring of the `WRAP_SINGLE_KEYS` option up. (Some bits were missed in the first PR)

### Testing done 

Usual

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOptions.java [hunk 1]
```diff
@@ -60,33 +59,22 @@ private SerdeOptions(final ImmutableSet<SerdeOption> options) {
...
-  public Optional<SerdeOption> keyWrapping() {
-    return Optional.ofNullable(
-        Iterables.getFirst(Sets.intersection(options, KEY_WRAPPING_OPTIONS), null)
-    );
-  }
-
-  public Optional<SerdeOption> valueWrapping() {
-    return Optional.ofNullable(
-        Iterables.getFirst(Sets.intersection(options, VALUE_WRAPPING_OPTIONS), null)
-    );
```
Comment (big-andy-coates) [inline ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOptions.java:89]: Dropped these in favour of `findAny`, which is more generic.

File: ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOptions.java [hunk 2]
```diff
@@ -111,6 +99,13 @@ public String toString() {
     return options.toString();
   }
 
+  private EnabledSerdeFeatures features(final boolean key) {
+    return EnabledSerdeFeatures.from(options.stream()
+        .filter(option -> option.isKeyOption() == key)
+        .map(SerdeOption::requiredFeature)
+        .collect(Collectors.toSet()));
+  }
```
Comment (big-andy-coates) [inline ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOptions.java:107]: key features are now returned as key features, not value features!

File: ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java [hunk 3]
```diff
@@ -38,19 +38,25 @@
...
    * Key version of {@link #UNWRAP_SINGLE_VALUES}.
    */
-  UNWRAP_SINGLE_KEYS(SerdeFeature.UNWRAP_SINGLES);
+  UNWRAP_SINGLE_KEYS(SerdeFeature.UNWRAP_SINGLES, true);
 
+  private final boolean isKey;
   private final SerdeFeature requiredFeature;
 
-  SerdeOption(final SerdeFeature requiredFeature) {
+  SerdeOption(final SerdeFeature requiredFeature, final boolean isKey) {
```
Comment (vcrfxia) [inline ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java:51]: Does it make sense to assume all serde options will always be either key- or value-specific, never both nor neither? Can't think of why this wouldn't be true in the future but seemed worth double-checking.

**Review by vcrfxia — APPROVED** [review]
LGTM. Thanks!

File: ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java (hunk 3 above)
Comment (big-andy-coates) [reply ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java:51]: That's the assumption at the moment.  If that changes... we change the code! ;)

Ideally, we'd ditch `SerdeOption` and just keep `SerdeFeature`, but that's a bigger change.

**Pull request merged**